src/ert/_c_wrappers/enkf/config/parameter_config.py

```diff
@@ -1,16 +1,29 @@
 from __future__ import annotations
 
+import dataclasses
 from abc import ABC, abstractmethod
-from dataclasses import dataclass
+from pathlib import Path
 from typing import TYPE_CHECKING
 
 if TYPE_CHECKING:
-    from pathlib import Path
+    from typing import Any
 
     from ert.storage import EnsembleAccessor, EnsembleReader
 
 
-@dataclass
+class CustomDict(dict):
```

Comment:
Perhaps a more explicit name is `PathStrDict`.

Reply:
The reason for the name is so we can add other custom conversions in the future

Reply:
A future that may never come, but I see your point.